src/config.rs

```diff
@@ -1,16 +1,21 @@
 #[derive(Debug, PartialEq, Eq)]
 pub struct Config {
-    pub remote_machine: RemoteMachine,
-    pub compression: Compression,
+    pub remote: Remote,
+    pub push: Push,
+    pub pull: Pull,
 }
 
 #[derive(Debug, Eq, PartialEq)]
-pub struct RemoteMachine {
+pub struct Remote {
     pub host: String,
 }
 
 #[derive(Debug, Eq, PartialEq)]
-pub struct Compression {
-    pub local: i64,
-    pub remote: i64,
+pub struct Push {
+    pub compression: i64,
+}
+
+#[derive(Debug, Eq, PartialEq)]
```

Comment:
Why do you need both `Eq` and `PartialEq`?

Reply:
That's a great question my friend!

I need equality for unit tests (`assert_eq!()`)
In Rust `Eq` seems to be special case of `PartialEq` and most of things like `assert_eq!()` need `PartialEq` thus I definitely need to  have `PartialEq`

Removed `Eq` 👍